### ISSUE TITLE
Install tiller per namespace as part of namespace creation

### ISSUE THREAD

**Issue by kalbir** [ministryofjustice/cloud-platform-test-cluster-env#7]
Estimate: 3

**Comment by nayoa** [comment]
https://medium.com/@amimahloof/how-to-setup-helm-and-tiller-with-rbac-and-namespaces-34bf27f7d3c3
https://github.com/ministryofjustice/k8s-nonprod-environments/blob/master/buildspec.yml
https://github.com/ministryofjustice/k8s-nonprod-environments/blob/master/namespaces/laa-fee-calculator-dev/rbac-config-laa-fee-dev.yaml

**Comment by nayoa** [comment]
After clarification with @kerin - we want to keep the existing tiller config. However, we may want to restrict it's access to our namespaces (kube-system, default etc) at a later date.

On the other hand, a tiller deploy should be included in the package when creating a new app namespace.

**Comment by nayoa** [comment]
https://github.com/ministryofjustice/k8s-nonprod-environments/pull/12

**Comment by nayoa** [comment]
As part of the namespace creation, an additional rbac-config has been added to the package. https://github.com/ministryofjustice/k8s-nonprod-environments/blob/master/namespaces/laa-fee-calculator-dev/rbac-config-laa-fee-dev.yaml

This is deployed during the codebuild stage and deploys a serviceaccount and a rolebinding that defines tiller has access to everything within the specified namespace. 

After deploying the rbac-config.yaml, a command is required to install tiller in the specified namespace. https://github.com/kubernetes/helm/blob/master/docs/rbac.md#example-deploy-tiller-in-a-namespace-restricted-to-deploying-resources-only-in-that-namespace

This command has been added to the namespace.py script. For every namespace in the cluster, it will run the command and change slightly depending on the name of the namespace. 

Example of how to use helm after tiller has been installed per namespace:

helm install stable/wordpress --tiller-namespace mps-dev --namespace mps-dev
helm list --tiller-namespace mps-dev

The tiller namespace is defined in the rbac-config. 

Just a 'helm list' on it's own will show everything installed in the 'kube-system' tiller namespace. 

After discussion with @kerin  we have found that you are able to set an environment variable for '--tiller-namespace' to avoid using it in every command. This environment variable is also mentioned in 'helm -h'

$TILLER_NAMESPACE

https://github.com/kubernetes/helm/issues/2519
